packages/good-design/src/theme/hoc/withTheme.tsx

```diff
@@ -16,10 +17,18 @@ export const withTheme =
     }
 
     const Wrapped: React.ComponentType<T> = ({ children, ...props }) => {
-      const themeProps = useThemeProps(id, props);
+      // useThemeProps expects themed props,
```

Comment:
@L03TJ3 I suggest the following solution

1. no `ntp_` prefix needed
2. by default all array props aren't considered as 'themed', they're passing directly to component
3. if we want to specify some array props to be 'themed' - we will pass their names **explicitly** to the HoC config object

Reply:
I followed the suggestion but I did choose to mirror it.
its native-base components and we are here in a withTheme hook, used for applying a theme.

So it makes more sense to expect the props to be themed over functional ones.

(this also means no refactoring is needed for existing components)